src/clib/Cabinet.h

```diff
@@ -61,12 +63,14 @@ class SharedCabinet
     /**
      * Add a new object. The index of the object is returned.
      */
-    static int add(shared_ptr<M> obj) {
+    static int add(shared_ptr<M> obj, int parent=-1) {
         dataRef data = getData();
         data.push_back(obj);
+        parentRef parents = getParents();
```

Comment:
You could just write:
```suggestion
        auto& parents = getParents();
```
and eliminate the need for the `parentRef` typedef.